### PR TITLE
Fix issue #1096

### PR DESCRIPTION
**Pull request by pu239ppy**
This is a fix for #1096

### CODE REVIEW

File: cookbooks/bcpc-hadoop/libraries/utils.rb [hunk 1]
```diff
@@ -338,7 +338,7 @@ def process_restarted_after_failure?(restart_failure_time, process_identifier)
   require 'time'
   begin
     start_time = process_start_time(process_identifier)
-    if start_time.nil? && (Time.parse(restart_failure_time).to_i < Time.parse(start_time).to_i)
+    if not start_time.nil? && (Time.parse(restart_failure_time).to_i < Time.parse(start_time).to_i)
```
Comment (vt0r) [inline cookbooks/bcpc-hadoop/libraries/utils.rb]: First, please prefer to use `!` over `not`, [due to possible issues with precedence](https://stackoverflow.com/questions/27429711/can-someone-explain-not-vs-in-ruby). Second, Rubocop is a fickle... uh... beast... that will also [fault you for checking to see if something is non nil](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/NonNilCheck), so maybe this should instead read as follows:

``` ruby
if start_time && (Time.parse(restart_failure_time).to_i < Time.parse(start_time).to_i)
```

**Comment by cbaenziger** [comment]
FWIW this code should be deprecated to use `locking_resource` using `hadoop_service` is not recommended.

File: cookbooks/bcpc-hadoop/libraries/utils.rb (hunk 1 above)
Comment (pu239ppy) [reply cookbooks/bcpc-hadoop/libraries/utils.rb]: The is what it originally was I believe

Comment (vt0r) [reply cookbooks/bcpc-hadoop/libraries/utils.rb]: @pu239ppy Ok so I found the two commits that modified this. The first one starts at this line:

https://github.com/bloomberg/chef-bach/commit/cf690ae7f5da6b9cf240b075bbbc01df8704c32a#diff-4301562bb3786f09a63001f8d6489b37L382

And the second one is the one you linked, at this line:

https://github.com/bloomberg/chef-bach/commit/a3c0d112c3a29c9d17096933fb4c7c956f2079ea#diff-4301562bb3786f09a63001f8d6489b37R341

It seems to me that simply making this PR change it to the following (same as my comment says above) will do exactly what we want, restoring the original condition as well:

``` ruby
if start_time ...
```

As for the `locking_resource` modification, I do agree it should be moved, but I'd say that's far out of scope of this PR.

**Pull request merged**